File: Conceitos de LPs/Tipos de Dados/Conjunto/conjunto.js
//JavaScript

// Definindo e inicializando conjuntos
const conjuntoA = new Set([1, 2, 3, 4, 5]);
const conjuntoB = new Set([4, 5, 6, 7, 8]);

// Função para união
function uniao(setA, setB) {
    return new Set([...setA, ...setB]);
}

// Função para intersecção
function intersecao(setA, setB) {
    return new Set([...setA].filter(x => setB.has(x)));
}

// Função para diferença
function diferenca(setA, setB) {
    return new Set([...setA].filter(x => !setB.has(x)));
}

// Exibindo resultados
console.log("União:", Array.from(uniao(conjuntoA, conjuntoB)));
console.log("Intersecção:", Array.from(intersecao(conjuntoA, conjuntoB)));
console.log("Diferença:", Array.from(diferenca(conjuntoA, conjuntoB)));

File: Conceitos de LPs/Tipos de Dados/Conjunto/conjunto.rs
//Rust

use std::collections::HashSet;

fn main() {
    // Definindo e inicializando conjuntos
    let conjunto_a: HashSet<i32> = [1, 2, 3, 4, 5].iter().cloned().collect();
    let conjunto_b: HashSet<i32> = [4, 5, 6, 7, 8].iter().cloned().collect();

    // União
    let uniao: HashSet<_> = conjunto_a.union(&conjunto_b).collect();

    // Intersecção
    let intersecao: HashSet<_> = conjunto_a.intersection(&conjunto_b).collect();

    // Diferença
    let diferenca: HashSet<_> = conjunto_a.difference(&conjunto_b).collect();

    // Exibindo resultados
    println!("União: {:?}", uniao);
    println!("Intersecção: {:?}", intersecao);
    println!("Diferença: {:?}", diferenca);
}
